pkg/kindsys/bind.go

```diff
@@ -0,0 +1,48 @@
+package kindsys
+
+import (
+	"github.com/grafana/thema"
+)
+
+var _ Composable = genericComposable{}
+
+type genericComposable struct {
+	decl Decl[ComposableProperties]
+	lin  thema.Lineage
+}
```

Comment:
In writing this PR, i glimpsed another significant simplification.

Most of the jennies written so far are based on `DeclForGen`, which are basically a validated `cue.Value` + a validated `thema.Lineage`. I originally called them that because i was thinking about them in contrast to [the `type Kind struct`](https://github.com/grafana/grafana/blob/a277d504a2b258d01be3a1aa401a19a33b8c817d/pkg/kinds/dashboard/dashboard_kind_gen.go#L24-L33) that we generate in `pkg/kinds/*/` and which implements [one of the main kindsys interfaces](https://github.com/grafana/grafana/blob/a277d504a2b258d01be3a1aa401a19a33b8c817d/pkg/kindsys/kind.go). What defines those structs is that they contain [references to the generated Go type](https://github.com/grafana/grafana/blob/a277d504a2b258d01be3a1aa401a19a33b8c817d/pkg/kinds/dashboard/dashboard_kind_gen.go#L26).

My error was in thinking that because the jennies needed something generic, that meant they needed something _other_ than an implementation of one of those main interfaces. But, as this file here demonstrates, that's actually exactly what they need. That means we can get rid of this "*ForGen" nonsense, and create parallel funcs for `BindCustom` and `BindCore` that correspond to structs like `genericComposable` here, and that'll be what generic parts of the system (including codegen, but also e.g. in the apiserver) actually depend on.

 reason to have a separate `DeclForGen`. Really, it would be easy to just have a single  core kinds are the exception here.  From the perspective of the  there's no reason to be thinking about

Reply:
@IfSentient you may find this of interest